tests/fsharpqa/Source/nunitConf.fs

```diff
@@ -0,0 +1,420 @@
+        { inherit TestCaseAttribute(arguments) }
+        then
+            this.Properties |> FSharpTestSuite.setProps dir
+            this.Properties.Set("ENVLSTLINE", envLstLine)
+
+
+open EnvLst
+
+type FSharpQASuiteTestAttribute(dir: string) =
+    inherit NUnitAttribute()
+
+    let _builder = NUnit.Framework.Internal.Builders.NUnitTestCaseBuilder()
+    interface NUnit.Framework.Interfaces.ITestBuilder with
+        member x.BuildFrom(methodInfo, suite) =
+            let rootDir = __SOURCE_DIRECTORY__
+            let path = Path.Combine(Path.Combine(rootDir, dir), "env.lst")
+
+            let lines =
+                path
+                |> IO.File.ReadAllLines
+                |> Array.map (fun l -> l, EnvLst.parseLine l)
+
+            let errors = lines |> Array.choose (function l, Choice2Of2 e -> Some(l,e) | _ -> None)
+            match errors with
+            | [| |] -> ()
+            | err ->
+                log "Invalid env.lst file '%s'" path
+                log "Errors:"
+                err |> Array.iter (fun (l,e) -> log "- %s. Line '%s'" e l)
+                failwithf "Invalid env.lst file '%s'. Errors: %A" path err
+
+            match lines |> Array.choose (function l, Choice1Of2(Some(EnvLstLine.Data(d))) -> Some(l,d) | _ -> None) with
+            | [| |] ->
+                failwithf "No valid lines inside env.lst file '%s'" path
+            | validLines ->
+                validLines
+                |> Array.map (fun (line,data) ->
+                    let tc = EnvLstLineTestCaseData (dir, line)
+                    
+                    //the id of test, it's the comment string
+                    let testName =
+                        data.Comment
+                        |> Option.map (fun s -> s.Trim())
+                        |> Option.map (fun s -> s.Replace(".", "_")) //test name is a full class name, the dot mean nesting
+                    
+                    testName
+                    |> Option.iter (fun testName -> tc.SetName(testName) |> ignore)
+
+                    _builder.BuildTestMethod(methodInfo, suite, tc) )
+                |> Seq.ofArray
+
+
+module FSharpQATestSuite =
+
+    let envLstData () = attempt {
+        let test = NUnit.Framework.TestContext.CurrentContext.Test
+        let line = test.Properties.Get("ENVLSTLINE") :?> string
+        
+        return! 
+            match line |> EnvLst.parseLine with
+            | Choice1Of2 (Some(Data(a))) -> succeed a.Vars
+            | Choice1Of2 (Some(Comment(c))) -> genericError (sprintf "The env.lst line '%s' is a comment" line)
+            | Choice1Of2 None -> genericError (sprintf "Invalid env.lst line '%s'" line)
+            | Choice2Of2 f -> genericError (sprintf "Invalid env.lst line '%s': %A" line f)
+
+        }
+
+module FileGuard =
+    let private remove path = if File.Exists(path) then Commands.rm (Path.GetTempPath()) path
+
+    [<AllowNullLiteral>]
+    type T (path: string) =
```

Comment:
ok. it's possible to use module instead of a class with static `Create(path)` method?
i really like the module syntax with functions :smile: .
i understand now `T` it's bad (it was a phase of my code style training i think, that's fsharp suite code, ~2 years ago), but i like module